library/std/src/sys/pal/unix/rand.rs

```diff
@@ -273,39 +246,30 @@ mod imp {
                     0,
                 )
             };
-            if ret == -1 || s_len != s.len() {
-                panic!(
-                    "kern.arandom sysctl failed! (returned {}, s.len() {}, oldlenp {})",
-                    ret,
-                    s.len(),
-                    s_len
-                );
+            if ret == -1 {
+                return Err(Error::last_os_error());
+            } else if s_len != s.len() {
+                // FIXME(joboet): this can't actually happen, can it?
```

Comment:
That surprised me too. But it's pre-existing and this code will presumably disappear with time in any case so I'm not too concerned about the `FIXME` ever being addressed.